packages/block-library/src/cover/edit.js

```diff
@@ -397,6 +397,15 @@ function CoverEdit( {
 	const isImageBackground = IMAGE_BACKGROUND_TYPE === backgroundType;
 	const isVideoBackground = VIDEO_BACKGROUND_TYPE === backgroundType;
 
+	useEffect( () => {
+		const defaultDimRatio =
+			isImageBackground || isVideoBackground ? 50 : 100;
+
+		setAttributes( {
+			dimRatio: 'number' === typeof dimRatio ? dimRatio : defaultDimRatio,
+		} );
+	}, [ isImageBackground, isVideoBackground ] );
```

Comment:
Let's include `dimRatio` in dependencies.

Setting attributes can cause an "undo trap." It's not visible with the Cover block, but it might be worth indicating that these attribute updates shouldn't create the "undo" history. More details: https://github.com/WordPress/gutenberg/pull/34565

Currently, I had to click three times to undo the Cover block insertion.

https://user-images.githubusercontent.com/240569/141464027-6624b3cf-810c-4d47-aaf2-b7231b11f172.mp4

Reply:
done, do you think we should move the withDispatch to a useDispatch as  well?

Reply:
Thanks, @glendaviesnz. I think that would be a good change. We're trying to slowly move away from HOCs.